mkosi/config.py

```diff
@@ -3184,6 +3221,14 @@ def create_argument_only_parser(chdir: bool = True) -> argparse.ArgumentParser:
         help="The format to show documentation in",
         default=DocFormat.auto,
         type=DocFormat,
+        choices=list(DocFormat)
+    )
+    parser.add_argument(
+        "--shell-completion",
```

Comment:
Hmm, `bin/mkosi --shell-completion` fails with `mkosi: error: argument --shell-completion: expected one argument`, and `bin/mkosi --shell-completion auto` starts a build. Oh, it's supposed to be used with `completion` verb. I think the verb is not needed, since we need the switch anyway. So I'd make `--shell-completion` standalone like `--help` or `--version`.